dfastmi/batch/AnalyserDflowfm.py

```diff
@@ -0,0 +1,376 @@
+        xykm : shapely.geometry.linestring.LineString
+            Original river chainage line.
+        n_fields : int
+            Number of fields to process (e.g. to cover a tidal period).
+        tide_bc : Tuple[str, ...]
+            Array of tidal boundary condition; one per forcing condition.
+        plotops : Dict
+            Dictionary of plot settings
+
+        Returns
+        -------
+        success : bool
+            Flag indicating whether analysis could be carried out.
+        Output data : OutputDataDflowfm
+            DTO with the data which is needed to create a report.
+        """
+        one_fm_filename, missing_data = self._get_first_fm_data_filename(q_threshold, discharges, rsigma, filenames, tide_bc)
+
+        if missing_data:
+            return missing_data, None
+
+        self._logger.log_load_mesh()
+        xn, yn, face_node_connectivity = self._get_xynode_connect(one_fm_filename)
+
+        xykm_data = self._get_xykm_data(xykm, xn, yn, face_node_connectivity)
+
+        if xykm is None and self._needs_tide:
+            self._logger.print_riverkm_needed_for_tidal()
+            return True, None
+
+        missing_data, dzq = self._get_dzq(discharges, rsigma, ucrit, filenames, n_fields, tide_bc, missing_data, xykm_data.iface, xykm_data.dxi, xykm_data.dyi)
+
+        if not missing_data:
+            self._logger.log_char_bed_changes()
+
+            if tstag > 0:
+                dzq = (dzq[0], dzq[0], dzq[1], dzq[2])
+                fraction_of_year = (fraction_of_year[0], tstag, fraction_of_year[1], fraction_of_year[2])
+                rsigma = (rsigma[0], 1.0, rsigma[1], rsigma[2])
+
+            # main_computation now returns new pointwise zmin and zmax
+            dzgemi, dzmaxi, dzmini, dzbi = main_computation(dzq, fraction_of_year, rsigma)
+
+            if self._old_zmin_zmax:
+                # get old zmax and zmin
+                dzmaxi = dzbi[0]
+                zmax_str = "maximum bed level change after flood without dredging"
+                dzmini = dzbi[1]
+                zmin_str = "minimum bed level change after low flow without dredging"
+            else:
+                zmax_str = "maximum value of bed level change without dredging"
+                zmin_str = "minimum value of bed level change without dredging"
+
+        sedimentation_data = None
+        if xykm is not None:
+            sedimentation_data = comp_sedimentation_volume(xykm_data.xni, xykm_data.yni, xykm_data.sni, xykm_data.nni, xykm_data.face_node_connectivity_index, dzgemi, slength, nwidth, xykm_data.xykline, self._outputdir, plotops)
+
+        return missing_data, OutputDataDflowfm(rsigma, one_fm_filename, xn, face_node_connectivity, dzq, dzgemi, dzmaxi, dzmini, dzbi, zmax_str, zmin_str, xykm_data, sedimentation_data)
+
+    def _get_first_fm_data_filename(self,q_threshold: float, discharges: Vector, rsigma: Vector, filenames: Dict[Any, Tuple[str,str]], tide_bc: Tuple[str, ...]) -> Tuple[str, bool]:
```

Comment:
are all arguments here needed, can't they be class fields

Reply:
I will look into it

Reply:
have a look at how it is now.